src/main/java/com/kuliginstepan/dadata/client/DadataClient.java

```diff
@@ -26,40 +26,28 @@
 import com.kuliginstepan.dadata.client.domain.postal.PostalOfficeRequest;
 import com.kuliginstepan.dadata.client.exception.DadataException;
 import com.kuliginstepan.dadata.client.exception.ErrorDetails;
-import io.netty.handler.timeout.ReadTimeoutHandler;
-import lombok.RequiredArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
 import org.springframework.core.ParameterizedTypeReference;
-import org.springframework.http.HttpHeaders;
-import org.springframework.http.client.reactive.ReactorClientHttpConnector;
 import org.springframework.web.reactive.function.BodyInserters;
 import org.springframework.web.reactive.function.client.ClientResponse;
 import org.springframework.web.reactive.function.client.WebClient;
 import reactor.core.publisher.Flux;
 import reactor.core.publisher.Mono;
-import reactor.netty.http.client.HttpClient;
-import reactor.netty.tcp.TcpClient;
 
-import java.time.Duration;
-import java.time.temporal.ChronoUnit;
-import java.util.concurrent.TimeUnit;
-
-@RequiredArgsConstructor
+/**
+ * Should be build using {@link DadataClientBuilder}
+ */
 @Slf4j
 public class DadataClient {
 
-    private static final String BASE_API_URL = "https://suggestions.dadata.ru/suggestions/api/4_1/rs";
     private static final String SUGGESTION_PREFIX = "/suggest";
     private static final String GEOLOCATE_PREFIX = "/geolocate";
     private static final String FIND_BY_ID_PREFIX = "/findById";
-    private static final Duration DEFAULT_TIMEOUT = Duration.of(5, ChronoUnit.SECONDS);
 
-    private final String token;
-    private final Duration timeout;
+    private final WebClient webClient;
 
-    public DadataClient(String token) {
-        this.token = token;
-        timeout = DEFAULT_TIMEOUT;
+    DadataClient(WebClient webClient) {
```

Comment:
Please make this constructor `protected`

Reply:
done